staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -82,28 +82,45 @@ type kmsv2PluginProbe struct {
 	l            *sync.Mutex
 }
 
+type kmsHealthChecker []healthz.HealthChecker
+
+func (k kmsHealthChecker) Name() string {
+	return "kms-provider-0" // for backwards compatibility with /healthz?exclude=kms-provider-0
+}
+
+func (k kmsHealthChecker) Check(req *http.Request) error {
+	for i := range k {
+		checker := k[i]
+		if err := checker.Check(req); err != nil {
+			return fmt.Errorf("%s healthz check failed: %w", checker.Name(), err)
+		}
```

Comment:
Should we check all the health checks and return an aggregate error?

Reply:
yes, we should not short-circuit and should surface the failures from multiple providers if they exist